client-src/elements/chromedash-feature-page.ts

```diff
@@ -128,11 +130,31 @@ export class ChromedashFeaturePage extends LitElement {
     return this.feature && Object.keys(this.feature).length !== 0;
   }
 
+  fetchClosestShippingDate(milestone: number) {
+    if (milestone === 0) {
+      return;
+    }
+    const milestonePromise = window.csClient.getSpecifiedChannels(
+      milestone,
+      milestone
+    );
+    Promise.all([milestonePromise])
+      .then(([newMilestonesInfo]) => {
+        this.closestShippingDate = newMilestonesInfo[milestone]?.final_beta;
+        this.hasShipped = true;
+      })
+      .catch(() => {
+        showToastMessage(
+          'Some errors occurred. Please refresh the page or try again later.'
+        );
+      });
+  }
```

Comment:
Any reason not to write this as an async function?

```suggestion
  async fetchClosestShippingDate(milestone: number) {
    if (milestone === 0) {
      return;
    }
    try {
      const newMilestonesInfo = await window.csClient.getSpecifiedChannels(
        milestone,
        milestone
      );
      this.closestShippingDate = newMilestonesInfo[milestone]?.final_beta;
      this.hasShipped = true;
    } catch {
      showToastMessage(
        'Some errors occurred. Please refresh the page or try again later.'
      );
    }
  }
```

You could even just return the closestShippingDate and let the caller handle assigning it into member variables.

Reply:
Done!